### ISSUE TITLE
Firefox won't offer to save password

### ISSUE THREAD

**Issue by gsantner** [thelounge/thelounge#1740]
Currently when loggin in with username and password there is no popup asking me if I want to remember the login data in my browser password directory. It would be great if lounge could setup the elements accordlingy so this works :).

**Comment by dgw** [comment]
I doubt it gets much clearer than ["user" and "password"](https://github.com/thelounge/lounge/blob/90861effb4d22f0e3415b653f0c339c5a7c15780/client/views/windows/sign_in.tpl). Those have been the field names for [at least three years](https://github.com/thelounge/lounge/blob/cb1effa3759a1329fbbfff047e70044d674a4290/client/index.html).

This sounds like a browser issue. Perhaps you (or another person using your browser profile) told it never to save passwords for "this site" (the Lounge instance in question)?

**Comment by gsantner** [comment]
hmm. seems to happen ff57 (on any profile, new ones too), but not on chromium. well.. 

Completly new profile: I don't know what github.com is doing different, but here I get the popup asking on ff57. On lounge instance there is no such popup.

**Comment by dgw** [comment]
I have an idea or two, but I can't play with it until later tonight or tomorrow. But it's helpful to know that you're seeing the issue in Firefox 57 specifically, thanks for that. :)

**Comment by dgw** [comment]
Just spent the better part of an hour playing with this, and what I've gotten out of it is that Firefox is "special" when it comes to auto-filling passwords. (tl;dr, can a maintainer update the title of this issue to something like "Firefox won't offer to save password"?)

Both my development instance of The Lounge and the production instance I run remotely offered to save my password in **Chrome** 62, but not in Firefox 57. (As a baseline, I also verified that Firefox **did** offer to save my password on another webapp, my ZNC admin interface.)

I've tried to pick this apart. The Lounge doesn't actually submit the form anywhere. The form `action=""`, so it actually goes nowhere. When submitting the sign-in form, I see `Content Security Policy: The page’s settings blocked the loading of a resource at http://127.0.0.1:9000/ (“form-action 'none'”).` in the JS console on both FF and Chrome. (The same message appears in production, not only when running from localhost.)

One of the developers who wrote Firefox's password manager [explained](https://dolske.wordpress.com/2013/08/20/on-firefoxs-password-manager/) (back in 2013, though it's _probably_ still accurate) that Firefox uses form submission to determine when to offer to save a login:

> This whole process is initiated by a form submission. If a site doesn’t actually submit a form (via form.submit() or \<button type=submit>), but just runs some JavaScript to process the login, the password manager won’t see it. And thus can’t offer to save a new/changed login for the user. (Note that this is easy for a site to work around — do your work in the form’s onsubmit, but return |false| to cancel the submission).

Though there is a "submit" event handler [defined by](https://github.com/thelounge/lounge/blob/489bb8e395b5e2c1f5900d7f6b7f453753be6e5a/client/js/socket-events/auth.js#L28) the auth module, which does indeed do its work and then `return false`, I can't find it in the Event Listeners tab when I inspect the form in Chrome (Firefox's dev tools don't have that tab, apparently). And it's possible that the CSP request block trumps FF's password manager detecting a submit event anyway. At any rate, even with [`signon.debug=true`](https://wiki.mozilla.org/Toolkit:Password_Manager/Debugging) in about:config, no `LoginManagerPrompter` events appeared in FF's browser log, only `nsLoginManager` and related processes looking for existing saved logins. It can see and try to fill in login forms with existing saved credentials; it just can't detect when to offer to save new credentials.

I'm burnt out on this for today, but I'm sure this can be fixed. If I can't find any more info on my own, I might pop onto the Firefox mailing list or something. Surely _someone_ else _somewhere_ has come across this issue in another single page webapp…
